src/transforms.jl

```diff
@@ -8,192 +8,53 @@
-function lla2ecef(lla::LLA, d::WGS84)
-    lat = lla.lat
-    lon = lla.lon
-    alt = lla.alt
-
-    N = d.a / sqrt(1 - d.e*d.e * sind(lat)^2)   # Radius of curvature (meters)
-
-    x = (N + alt) * cosd(lat) * cosd(lon)
-    y = (N + alt) * cosd(lat) * sind(lon)
-    z = (N * (1 - d.e*d.e) + alt) * sind(lat)
-
-    return ECEF(x, y, z)
-end
-
 # For dictionary of nodes
-function lla2ecef(nodes::Dict{Int,LLA})
-    nodesECEF = Dict{Int,ECEF}()
-    datum = WGS84()
+function Geodesy.ECEF(nodes::Dict{Int,LLA}, datum::Ellipsoid = WGS84)
+    r = Dict{Int,ECEF}()
+    sizehint!(r, ceil(Int, 1.5*length(nodes)))
 
     for (key, node) in nodes
-        nodesECEF[key] = lla2ecef(node, datum)
+        r[key] = ECEF(node, datum)
     end
 
-    return nodesECEF
+    return r
 end
 
 ###############################################
 ### Conversion from ECEF to LLA coordinates ###
 ###############################################
 
-# For single-point calculations
-function ecef2lla(ecef::ECEF)
-    datum = WGS84() # Get WGS84 datum
-
-    return ecef2lla(ecef, datum)
-end
-
-# For multi-point loops
-function ecef2lla(ecef::ECEF, d::WGS84)
-    x = ecef.x
-    y = ecef.y
-    z = ecef.z
-
-    p = sqrt(x*x + y*y)
-    theta = atan2(z*d.a, p*d.b)
-    lambda = atan2(y, x)
-    phi = atan2(z + d.e_prime^2 * d.b * sin(theta)^3, p - d.e*d.e*d.a*cos(theta)^3)
-
-    N = d.a / sqrt(1 - d.e*d.e * sin(phi)^2)   # Radius of curvature (meters)
-    h = p / cos(phi) - N
-
-    return LLA(phi*180/pi, lambda*180/pi, h)
-end
-
-# For dictionary of nodes
-function ecef2lla(nodes::Dict{Int,ECEF})
-    nodesLLA = Dict{Int,LLA}()
-    datum = WGS84()
-
-    for (key, node) in nodes
-        nodesLLA[key] = ecef2lla(node, datum)
-    end
-
-    return nodesLLA
-end
-
-###############################################
-### Conversion from ECEF to ENU coordinates ###
-###############################################
-
-# Given a reference point for linarization
-function ecef2enu(ecef::ECEF, lla_ref::LLA)
-    # Reference point to linearize about
-    phi = lla_ref.lat
-    lambda = lla_ref.lon
-
-    ecef_ref = lla2ecef(lla_ref)
-    ecef_vec = [ecef.x - ecef_ref.x; ecef.y - ecef_ref.y; ecef.z - ecef_ref.z]
-
-    # Compute rotation matrix
-    R = [-sind(lambda) cosd(lambda) 0;
-         -cosd(lambda)*sind(phi) -sind(lambda)*sind(phi) cosd(phi);
-         cosd(lambda)*cosd(phi) sind(lambda)*cosd(phi) sind(phi)]
-    ned = R * ecef_vec
-
-    # Extract elements from vector
-    east = ned[1]
-    north = ned[2]
-    up = ned[3]
-
-    return ENU(east, north, up)
-end
-
-# Given Bounds object for linearization
-function ecef2enu(ecef::ECEF, bounds::Bounds{LLA})
-    lla_ref = centerBounds(bounds)
-
-    return ecef2enu(ecef, lla_ref)
-end
-
 # For dictionary of nodes
-function ecef2enu(nodes::Dict{Int,ECEF}, bounds::Bounds{LLA})
-    nodesENU = Dict{Int,ENU}()
-    lla_ref = centerBounds(bounds)
+function Geodesy.LLA(nodes::Dict{Int,ECEF}, datum::Ellipsoid = WGS84)
```

Comment:
(as with ENU) is there a reason why this function isn't moved into Geodesy.jl as well?